sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,507 @@
+"""Adapter for using the huggingface PatchTST for forecasting."""
+
+# documentation for PatchTST:
+# https://huggingface.co/docs/transformers/main/en/model_doc/patchtst#transformers.PatchTSTConfig
+
+import numpy as np
+import pandas as pd
+from skbase.utils.dependencies import _check_soft_dependencies
+
+from sktime.forecasting.base import ForecastingHorizon, _BaseGlobalForecaster
+from sktime.split import temporal_train_test_split
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    from torch.utils.data import Dataset
+else:
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+        pass
+
+
+if _check_soft_dependencies("transformers", severity="none"):
+    from transformers import (
+        PatchTSTConfig,
+        PatchTSTForPrediction,
+        PatchTSTModel,
+        Trainer,
+        TrainingArguments,
+    )
+
+
+class HFPatchTSTForecaster(_BaseGlobalForecaster):
+    """Interface for the PatchTST forecaster.
+
+    Paper: https://arxiv.org/abs/2211.14730
+
+    This model has 3 available modes:
+        1) Loading an untrained PatchTST model and pretrain on some dataset.
+        This can be done by passing in a PatchTST config dictionary or passing
+        in available parameters in this estimator, or by loading the
+        HFPatchTSTForecaster without any passed arguments
+        2) Load a pre-trained model for fine-tuning. The parameter `model_path`
+        can be load to load a pre-trained model for fine-tuning. Using a
+        pretrained model will override any other model config parameters, and
+        you can use the `fit` function to then fine-tune your pretrained model.
+        See the `model_path` docstring for more details.
+        3) Load a pre-trained model for zero-shot forecasting. The parameter
```

Comment:
skip a line here to render the numbered list correctly

you can see how the docs are rendered here: https://sktime--6787.org.readthedocs.build/en/6787/api_reference/auto_generated/sktime.forecasting.hf_patchtst_forecaster.HFPatchTSTForecaster.html

there might be some other formatting issues at other places